lib/client-options.js

```diff
@@ -33,7 +33,10 @@ function defaultOptions () {
     },
     pooling: {
       heartBeatInterval: 30000,
-      warmup: true
+      warmup: true,
+      // TODO: Make it protocol version dependent
+      // TODO: Use min value between set by the user and the protocol maximum
+      maxRequestsPerConnection: 2048
```

Comment:
It would be nice if this were set to 1024 for v3+ to match the java driver (unless you have a good reason for it to be 2048)

Another thing the java driver allows you to do is to set max requests per connection by host distance.  To be honest, I don't find that very useful and I don't think we should allow that here.